### DISCUSSION TITLE
is there a way to let `ruff check --watch` honor the `--format=json` option?

### DISCUSSION THREAD

**Discussion by haolian9** [astral-sh/ruff#3927] [Q&A] [answered]
hi folks! I found that `ruff check --format=json --watch foo.py` keep outputing `text` format messages rather than the asked format, `json`. do you think it's a valid use-case? also since the output json contains `\n`, would it be a problem in this use-case?

**Comment by charliermarsh** [comment, marked as answer]
It's not supported right now but I suppose we could support it. What's the use-case? Do you just prefer to have JSON output in your terminal? Or are you trying to pipe to another tool?

**Reply by haolian9** [reply]
hi, i was planning to get ruff and neovim married: a) spawn a ruff process along the side of neovim, b) keep reading from its stdout/stderr, c) parse the output as diagnostics, d) feed into nvim's diagnostic.

the things i concerned about in the above plan:
* cheap&fast: the ruff binary is about 14M on my linux machine, so a long-lived process `--watch` is good to have, yet I find no way to add new file to ruff's watching list.
* easy parsing: there is &errorformat in nvim, but it's so complicated that i dont want to use it. and json is prefered over plain text which is structual, though it has a overhead on (de)serializing.

there is a ruff-ls already, it requires python and several packages to run, i personally dont like that; but it made me want a C API from ruff, so that I can use it with luajit's FFI.

i'm not sure if what would the plan be resulted as: a new nvim plugin to use ruff or integrating it into nvim-null-ls or [efm-server](https://github.com/mattn/efm-langserver) or something else.

it's just a random thought, thank you for taking time on it!
